app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -174,50 +147,68 @@
+          apiEventsWatcher = null;
         }
       };
 
-      var render = function() {
-        $rootScope.$evalAsync(function () {
-          countUnreadNotificationsForAllGroups();
-          // NOTE: we are currently only showing one project in the drawer at a
-          // time. If we go back to multiple projects, we can eliminate the filter here
-          // and just pass the whole array as notificationGroups.
-          // if we do, we will have to handle group.open to keep track of what the
-          // user is viewing at the time & indicate to the user that the non-active
-          // project is "asleep"/not being watched.
-          drawer.notificationGroups = _.filter(notificationGroups, function(group) {
-            return group.project.metadata.name === $routeParams.project;
-          });
-        });
+      var deregisterNotificationListener = function() {
+        notificationListener && notificationListener();
+        notificationListener = null;
       };
 
-      // TODO: follow-on PR to decide which of these events to toast,
-      // via config in constants.js
-      var eventWatchCallback = function(eventData) {
-        eventsMap = formatAndFilterEvents(eventData.by('metadata.uid'));
-        // TODO: Update to an intermediate map, so that we can then combine both
-        // events + notifications into the final notificationGroups output
-        notificationGroups = sortNotificationGroups(eventsMap);
+      var apiEventWatchCallback = function(eventData) {
+        apiEventsMap[$routeParams.project] = formatAPIEvents(filterAPIEvents(eventData.by('metadata.name')));
         render();
       };
 
-      // TODO: Follow-on PR to update & add the internal notifications to the
-      // var notificationWatchCallback = function(event, notification) {
-      //   // will need to add .event = {} and immitate structure
-      //   if(!notification.lastTimestamp) {
-      //     // creates a timestamp that matches event format: 2017-08-09T19:55:35Z
-      //     notification.lastTimestamp = moment.parseZone(new Date()).utc().format();
-      //   }
-      //   clientGeneratedNotifications.push(notification);
-      // };
-
-      var iconClassByEventSeverity = {
-        Normal: 'pficon pficon-info',
-        Warning: 'pficon pficon-warning-triangle-o'
+      var notificationWatchCallback = function(event, notification) {
+        if(!notification.showInDrawer) {
+          return;
+        }
+        var project = notification.namespace || $routeParams.project;
+        var id = notification.id || _.uniqueId('notification_') + Date.now();
+        notificationsMap[project] = notificationsMap[project] || {};
+        notificationsMap[project][id] = {
+          actions: null,
+          unread: !EventsService.isRead(id),
+          // using uid to match API events and have one filed to pass
+          // to EventsService for read/cleared, etc
+          trackByID: notification.trackByID,
+          uid: id,
```

Comment:
I don't think we need to set this anymore? Technically it's not a unique ID since `notification.id` is not necessarily unique.